rosetta-testing/src/test/java/com/regnosys/rosetta/types/SubtypeRelationTest.xtend

```diff
@@ -28,6 +36,166 @@ class SubtypeRelationTest {
 		return model.elements.filter(RosettaNamed).findFirst[it.name == name] as RosettaEnumeration
 	}
 	
+	@Test
+	def testJoinOnMetaSubTypesReturnsParent() {
+		val model = '''
+			type A:
+			
+			type B extends A:
+			
+			type C extends A:
+				
+		'''.parseRosetta
+		
+		val fieldB = '''
+			fieldB B (1..1)
+				[metadata reference]
+		'''.parseAttribute(#[model]).RTypeOfSymbol
+		
+		val fieldC = '''
+			fieldC C (1..1)
+				[metadata reference]
+		'''.parseAttribute(#[model]).RTypeOfSymbol
+		
+		
+		val fieldA = model.getData("A").buildRDataType
+		
+		val joined = fieldB.join(fieldC)
+		
+		assertEquals(joined.RType, fieldA)
+		assertFalse(joined.hasMeta)
+	}
+	
+	@Test
+	def testJoinMetaTypeWithNonMetaTypeIsCorrect() {
+		val fieldA = '''
+			fieldA number (1..1)
+				[metadata scheme]
+				[metadata reference]
+		'''.parseAttribute.RTypeOfSymbol
+		
+		val fieldB = '''
+			fieldB number (1..1)
+		'''.parseAttribute.RTypeOfSymbol
+		
+		val result = fieldA.join(fieldB)
+		assertTrue(result.RType instanceof RNumberType)
+		assertFalse(result.hasMeta)
+	}
+	
+	@Test
+	def testJoinOnSameBaseTypeWithMetaIsCorrect() {
+		val fieldA = '''
+			fieldA string (1..1)
+				[metadata scheme]
+				[metadata reference]
+		'''.parseAttribute.RTypeOfSymbol
+		
+		val fieldB = '''
+			fieldB string (1..1)
+				[metadata scheme]
+				[metadata address]
+		'''.parseAttribute.RTypeOfSymbol
+				
+		val result = fieldA.join(fieldB)
+		assertTrue(result.RType instanceof RStringType)
+		assertEquals(#[new RMetaAttribute("scheme", UNCONSTRAINED_STRING)], result.getMetaAttributes)
+	}
```

Comment:
I think this test tests the same code paths as `testJoinMetaTypeWithNonMetaTypeIsCorrect` and `testStringWithSchemeAndReferenceHasNoSubtypeRelationWithStringWithAddressAndLocation` - I wonder if we need all three of them. (potential of unnecessary maintenance)